### ISSUE TITLE
InvalidCastException in NumericUpDown

### ISSUE THREAD

**Issue by nsrahmad** [AvaloniaUI/Avalonia#9848]
I am learning so I don't know If I am doing something wrong or it is a bug.

**Describe the bug**
NumericUpDown shows System.InvalidCastException: Unable to convert object '89.6' of type 'System.Double' to type 'System.Nullable'1[System.Decimal]'.
I am using a class implementing `IValueConverter` to convert the value.

**To Reproduce**
Project has just a few lines changed from template:
https://github.com/nsrahmad/7GUI-Avalonia/tree/master/TemperatureConverter

**Expected behavior**
No errors. The error contains the right answer, so conversion is happening.

**Screenshots**
![TempConv](https://user-images.githubusercontent.com/10996250/210311605-2bc93c48-25e5-4de7-9567-efabba7be07c.gif)

**Desktop (please complete the following information):**

- OS: Windows 10
- Version 11.0.0-preview4

**Additional context**

**Comment by timunie** [comment]
@nsrahmad I think we had a breaking change here in 11.0. The reason is, that `double` has rounding issues where `decimal` doesn't have these problems. This is important if you work with percentage or currency values.

I see two options for you: 
1. use `decimal` in the bound Property
2. use a `Converter` 

See also: https://github.com/AvaloniaUI/Avalonia/pull/5981

**Comment by maxkatz6** [comment]
Not related but you also have a typo here
![image](https://user-images.githubusercontent.com/3163374/210413047-56e5e621-0e09-4225-aee1-fd942a146649.png)

**Comment by nsrahmad** [comment]
@maxkatz6 Thanks.

@timunie Using `decimal` sort of worked. But I started to get StackOverflowException if I try to change the value with mouse wheel 3 or 4 times.

It took me some time to remove `FormatString` to uncover rounding issues in my converter. So trying to convert 1 celsius to fahrenheit would give:
```csharp
> Console.WriteLine((1M * (9M / 5M)) + 32M);
33.8
```
But trying to convert back `33.8` to celsius would give:
```csharp
> Console.WriteLine((33.8M - 32M) * (5M / 9M));
1.0000000000000000000000000001
```
I had to call `decimal.Round` on the result to fix it. Thank you for your help.
